packages/babel-generator/src/printer.js

```diff
@@ -58,7 +58,7 @@ export default class Printer extends Buffer {
     if (needsParens) this.push(")");
 
     // end
-    this.map.mark(node, "end");
+    if (parent) this.map.mark(parent);
```

Comment:
Yes, this was introduced because the source maps were not treating the output a tree causing incorrect mappings. I can't really provide much more information without knowing what issue this caused you or how this is worse, but if you tag me on the PR I will provide whatever feedback that I can.